### PR TITLE
feat(profiling): allow to pass service_name to the Profiler object

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

**Comment by jd** [comment]
That's a good suggestion, I've added it. I also renamed the attribute to `service` as I noticed the tracer used `service` and not `service_name`.

File: ddtrace/profiling/profiler.py [hunk 1]
```diff
@@ -32,6 +32,13 @@ def _build_default_exporters():
     return exporters
 
 
+def _get_service_name():
+    for service_name_var in ("DD_SERVICE", "DD_SERVICE_NAME", "DATADOG_SERVICE_NAME"):
+        service_name = os.environ.get(service_name_var)
+        if service_name is not None:
+            return service_name
+
```
Comment (Kyle-Verhoog) [inline ddtrace/profiling/profiler.py:40]: Are we okay with relying on the implicit `return None` here for the service name if one is not found? It was like this formerly so just a non-blocking thought.

File: tests/profiling/test_profiler.py [hunk 2]
```diff
@@ -16,3 +19,30 @@ def test_restart():
...
+        pytest.fail("Unable to find HTTP exporter")
+
+
+def test_service_api(monkeypatch):
+    monkeypatch.setenv("DD_API_KEY", "foobar")
+    prof = profiler.Profiler(service="foobar")
+    assert prof.service == "foobar"
+    for exporter in prof.exporters:
+        if isinstance(exporter, http.PprofHTTPExporter):
+            assert exporter.service_name == "foobar"
```
Comment (Kyle-Verhoog) [inline tests/profiling/test_profiler.py:45]: Should we also do `service_name` -> `service` in the exporter?

File: ddtrace/profiling/profiler.py (hunk 1 above)
Comment (jd) [reply ddtrace/profiling/profiler.py:40]: Yes, that's handled in the http profiler.

File: tests/profiling/test_profiler.py (hunk 2 above)
Comment (jd) [reply tests/profiling/test_profiler.py]: done

**Pull request merged**